scanpy/tools/_leiden.py

```diff
@@ -141,18 +141,43 @@ def leiden(
         categories=natsorted(np.unique(groups).astype('U')),
     )
     # store information on the clustering parameters
-    adata.uns['leiden'] = {}
-    adata.uns['leiden']['params'] = dict(
+    uns_key = key_added if key_added.startswith('leiden') else f'leiden_{key_added}'
```

Comment:
Why `f"leiden_{key_added}"` instead of `key_added`

Reply:
Because people typically use the leiden prefix e.g. key_added="leiden_2.0".

Reply:
I guess that’s the point I’m trying to make. People will specify the key they want to use, so I don’t think we should modify it. For example, if I want to use the params from a leiden call in some code I write, I’d prefer:

```python
sc.tl.leiden(adata, key_added=key_added, ...)
adata.uns[key_added][“quality”]
```

over

```python
sc.tl.leiden(adata, key_added=key_added, ...)
uns_key = key_added
if not key_added.startswith(“leiden”):
    uns_key = f“leiden_{uns_key}”
adata.uns[uns_key][“quality”]
```

Basically, if I wanted the key_added to start with “leiden”, I would’ve had the key added start with “leiden”.